src/originalBudgetTrackingApp/commonTypes.ts

```diff
@@ -10,7 +10,8 @@ export interface EnrichedTransaction extends Transaction {
 
 export enum OutputVendorName {
   YNAB = 'ynab',
-  GOOGLE_SHEETS = 'googleSheets'
+  GOOGLE_SHEETS = 'googleSheets',
+  JSON = 'json'
```

Comment:
I think we need to remove this enum and use the types from `configManager` because they are all created from one type definition:
```ts
export type OutputVendorsNames = keyof Config['outputVendors']
export type OutputVendorsConfigs = Exclude<Config['outputVendors'][OutputVendorsNames], undefined>
export type OutputVendorsConfig<T extends OutputVendorsNames> = Exclude<Config['outputVendors'][T], undefined>
```

Reply:
That is not possible as is, because you cannot use a type as a value. For that you need an enum. 

What i will do is move the enum into the configManager and use the enum values as the keys for the outputVendor config like so:

```
export interface Config {
  outputVendors: {
    [OutputVendorName.GOOGLE_SHEETS]?: GoogleSheetsConfig;
    [OutputVendorName.YNAB]?: YnabConfig;
    [OutputVendorName.JSON]?: JsonConfig;
  };
...
```

Reply:
OK then